WP_Auth0.php

```diff
@@ -436,27 +436,50 @@ public static function uninstall() {
 		delete_transient( WPA0_JWKS_CACHE_TRANSIENT_NAME );
 	}
 
+	/**
+	 * Look for a class within a specific set of paths.
+	 *
+	 * @param string $class - Class name to look for.
+	 *
+	 * @return bool
+	 */
 	private function autoloader( $class ) {
-		$path = WPA0_PLUGIN_DIR;
-		$paths = array();
-		$exts = array( '.php', '.class.php' );
-
-		$paths[] = $path.'lib/';
-		$paths[] = $path.'lib/admin/';
-		$paths[] = $path.'lib/exceptions/';
-		$paths[] = $path.'lib/wizard/';
-		$paths[] = $path.'lib/initial-setup/';
-		$paths[] = $path.'lib/twitter-api-php/';
-		$paths[] = $path.'lib/php-jwt/Exceptions/';
-		$paths[] = $path.'lib/php-jwt/Authentication/';
-		$paths[] = $path;
-
-		foreach ( $paths as $p ) {
-			foreach ( $exts as $ext ) {
-				if ( file_exists( $p.$class.$ext ) ) {
-					require_once $p.$class.$ext;
-					return true;
-				}
+		$source_dir = WPA0_PLUGIN_DIR . 'lib/';
+
+		// Catch non-name-spaced classes that still need auto-loading.
+		switch ( $class ) {
+			case 'TwitterAPIExchange':
+				require_once $source_dir . 'twitter-api-php/' . $class . '.php';
+				return true;
+
+			case 'JWT':
+				require_once $source_dir . 'php-jwt/Authentication/' . $class . '.php';
+				return true;
+
+			case 'BeforeValidException':
+			case 'ExpiredException':
+			case 'SignatureInvalidException':
+				require_once $source_dir . 'php-jwt/Exceptions/'  . $class . '.php';
+				return true;
+		}
+
+		// Anything that's not part of the above and not name-spaced can be skipped.
+		if ( 0 !== strpos( $class, 'WP_Auth0' ) ) {
+			return false;
+		}
+
+		$paths = array(
+			$source_dir,
+			$source_dir . 'admin/',
+			$source_dir . 'exceptions/',
+			$source_dir . 'wizard/',
+			$source_dir . 'initial-setup/',
```

Comment:
I guess this last comma will be ignored.

Reply:
Yes. It's a PHP standard to add a comma on the last line of an array